crates/sui-graphql-client/src/lib.rs

```diff
@@ -327,27 +329,42 @@ impl Client {
     }
 
     /// Internal function to handle pagination filters and return the appropriate values.
+    /// If limit is omitted, it will use the max page size from the service config.
     async fn pagination_filter(
         &self,
         pagination_filter: PaginationFilter,
     ) -> (Option<String>, Option<String>, Option<i32>, Option<i32>) {
-        let limit = if let Some(limit) = pagination_filter.limit {
-            limit
-        } else {
-            let cfg = self.service_config().await;
-            if let Ok(cfg) = cfg {
-                cfg.max_page_size
-            } else {
-                DEFAULT_ITEMS_PER_PAGE
-            }
-        };
+        let limit = pagination_filter
+            .limit
+            .unwrap_or(self.max_page_size().await.unwrap_or(DEFAULT_ITEMS_PER_PAGE));
```

Comment:
I hate that clippy forces nowadays to use `unwrap_or`, instead of `unwrap_or_else` for lazily evaluating this statement.